assets/migrations/mysql/004_new_user_parts_columns.sql

```diff
@@ -0,0 +1,103 @@
+-- This migration adds 3 more columns to the tuple and changelog tables, and fills past and future data with a trigger.
+
+-- +goose Up
+
+-- calculation for sizes are below
+ALTER TABLE tuple
+    ADD COLUMN user_object_type VARCHAR(145),
+    ADD COLUMN user_object_id VARCHAR(145),
+    ADD COLUMN user_relation VARCHAR(145);
+
+-- jon becomes (..., jon, ...)
+-- user:jon becomes (user, jon, ...)
+-- team:* becomes (team, *, ...)
+-- group:eng#member becomes (group, eng, member)
+UPDATE tuple SET
+         user_object_type = (CASE
+                                 WHEN position(':' in _user) > 0 THEN substring_index(_user, ':', 1)
+                                 ELSE ''
+             END),
+         user_object_id = (CASE
+                               WHEN position('#' in _user) > 0 THEN substring_index(substring_index(_user, ':', -1), '#', 1)
+                               ELSE substring_index(_user, ':', -1)
+             END),
+         user_relation = (CASE
+                              WHEN position('#' in _user) > 0 THEN substring_index(_user, '#', -1)
+                              ELSE ''
+             END);
+
+CREATE TRIGGER migrate_user_column_tuple BEFORE INSERT ON tuple
```

Comment:
It seems these triggers don't work like we expect.

Steps to reproduce the issue I am seeing:
1. Start Postgres
    `make start-postgres`
2. Migrate up to the latest schema version (version 3)
    `make migrate-postgres`
3. Run the latest version (v1.1.0) of OpenFGA locally
    `make run-postgres`
4. Write some tuples
5. Checkout this branch and run the migration for schema revision 4
    ```
    git checkout feat-split-user-column
    make migrate-postgres
    ```
6. Write another set of tuples


What I see is that the migration up to revision 4 (step 5) migrates existing tuples into the new user columns, but new writes (step 6) do not cause the `migrate_user_column_tuple` trigger. I believe the intent was for this trigger to cause updates to the new columns when values are written to the old columns (e.g. due to old OpenFGA replicas still being deployed and writing to the old columns).